lib/web/apiserver.go

```diff
@@ -1844,10 +1844,13 @@ func (h *Handler) getWebConfig(w http.ResponseWriter, r *http.Request, p httprou
 	automaticUpgradesEnabled := clusterFeatures.GetAutomaticUpgrades()
 	var automaticUpgradesTargetVersion string
 	if automaticUpgradesEnabled {
-		automaticUpgradesTargetVersion, err = h.cfg.AutomaticUpgradesChannels.DefaultVersion(r.Context())
+		const group, updaterUUID = "", ""
+		agentVersion, err := h.autoUpdateAgentVersion(r.Context(), group, updaterUUID)
 		if err != nil {
-			h.logger.ErrorContext(r.Context(), "Cannot read target version", "error", err)
+			h.logger.ErrorContext(r.Context(), "Cannot read autoupdate target version", "error", err)
 		}
+		// agentVersion doesn't have the leading "v" which is expected here.
+		automaticUpgradesTargetVersion = fmt.Sprintf("v%s", agentVersion)
```

Comment:
I wonder if this is correct because using the left hand side var when there's an error is asking for trouble.

Reply:
Yeah that's dangerous, I fixed it